ember-app/app/services/filter_groups/milestone.js

```diff
@@ -1,29 +1,50 @@
 import Ember from 'ember';
 
+function attr(modelProp, map) {
+  return Ember.computed("model." + modelProp, {
+    get: function(key){
+      var filters = this.get("model." + modelProp).map(map);
+      filters.insertAt(0, Ember.Object.create({
+        name: 'No milestone',
+        queryParam: "milestone",
+        mode:0,
+        condition:function(i){
+          return i.milestone == null;
+        }
+      }));
+      if(this._filters){
+        this._filters.forEach(function(f){
+          var newOne = filters.findBy('name', f.name);
+          if(newOne){
+            newOne.set('mode', f.mode);
+          }
+        });
+      }
```

Comment:
I have never tested it, but as a callback after we create something like a milestone could we not just do something like `filters.milestoneFilters.pushObject(newMilestoneFilter)` ? It would definitely be less reactive and more dependent on knowledge of the filter internals I suppose.